### PR TITLE
feat: add socket write status

### PR DESCRIPTION
**Pull request by leoner**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

### CODE REVIEW

File: lib/base.js [hunk 1]
```diff
@@ -201,7 +201,7 @@ class TCPBase extends Base {
       endTime: -1,
     };
     let endTime;
-    this._socket.write(packet.data, () => {
+    meta.succ = this._socket.write(packet.data, () => {
```
Comment (leoner) [inline lib/base.js]: Returns true if the entire data was flushed successfully to the kernel buffer. Returns false if all or part of the data was queued in user memory. 'drain' will be emitted when the buffer is again free.

File: lib/base.js [hunk 2]
```diff
@@ -409,7 +409,7 @@ class TCPBase extends Base {
       this._heartbeatTimer = setInterval(() => {
         const duration = this._lastHeartbeatTime - this._lastReceiveDataTime;
         if (this._lastReceiveDataTime && duration > this.options.heartbeatInterval) {
-          const err = new Error(`server no response in ${duration}ms, maybe the socket is end on the other side.`);
+          const err = new Error(`server ${this.options.host}:${this.options.port} no response in ${duration}ms, maybe the socket is end on the other side.`);
```
Comment (leoner) [inline lib/base.js]: 现在心跳的超时错误, 看不出来是那台服务器.

File: lib/base.js (hunk 1 above)
Comment (fengmk2) [inline lib/base.js]: `meta.writeSuccess` will be better

Comment (leoner) [reply lib/base.js]: done!

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
1.1.0

**Pull request merged**